### PR TITLE
Switch to uwsgi instead of gunicorn

### PR DESCRIPTION
**Pull request by waghanza**
Hi @murilobr,

This `PR` switch from `gunicorn` to `uwsgi`

The performance are slightly the same, but `uwsgi` is referenced in `cherrypy` doc :stuck_out_tongue: 

@webknjaz Do you approve this change ?

Regards,

PS : Do not merge until approval

### CODE REVIEW

File: python/cherrypy/config.yaml [hunk 1]
```diff
@@ -3,9 +3,8 @@ framework:
...
 
 command: >
-  gunicorn  \
-    --log-level warning  \
-    --bind 0.0.0.0:3000  \
-    --reuse-port  \
-    --workers $(nproc)  \
-      server:app
+  uwsgi \
+    --socket 0.0.0.0:3000 \
```
Comment (webknjaz) [inline python/cherrypy/config.yaml:7]: Maybe also add
```suggestion
    --socket 0.0.0.0:3000 \
    --async 100 \
```

**Comment by webknjaz** [comment]
I think it's ok

File: python/cherrypy/config.yaml (hunk 1 above)
Comment (waghanza) [reply python/cherrypy/config.yaml:7]: what does it means ?

Comment (murilobr) [reply python/cherrypy/config.yaml:7]: I'm waiting for this to approve.

Comment (webknjaz) [reply python/cherrypy/config.yaml:7]: It makes uswgi have async cores on its side.

Comment (waghanza) [reply python/cherrypy/config.yaml:7]: ah ok, but why `100` ?

Comment (murilobr) [reply python/cherrypy/config.yaml:7]: Maybe it should be the same variable used in other samples $(nproc)

Comment (webknjaz) [reply python/cherrypy/config.yaml:7]: No, it's not about threads. AFAIK there's one thread for I/O loop, it's about concurrency, the concept unrelated to CPU cores.

Comment (webknjaz) [reply python/cherrypy/config.yaml:7]: https://uwsgi-docs.readthedocs.io/en/latest/Async.html
basically, this number is the maximum concurrent HTTP requests taken by uwsgi

Comment (webknjaz) [reply python/cherrypy/config.yaml:7]: I tried out 1024 locally, which seems to be not loading my laptop at all. So maybe some adjustments will be needed.

Comment (waghanza) [reply python/cherrypy/config.yaml:7]: I see ... This is to adjust concurrency level

But :

   + in **production** we **CAN NOT** know in advance concurrency level
   + what problem arrive if I reach up this level ?

Comment (waghanza) [reply python/cherrypy/config.yaml:7]: more, I have `*** DANGER *** async mode without coroutine/greenthread engine loaded !!!` when running with `async` switch

Comment (webknjaz) [reply python/cherrypy/config.yaml:7]: That's fine. IRL you have to plan the load for production. I believe, if you reach the concurrency level, the new connections will be reset or kept hanging.

Comment (waghanza) [reply python/cherrypy/config.yaml:7]: on my side, I have :

| async <number> | latency average |
|:---------------------|:-----------------------|
| 1000 | 102.97 ms |
| 100 |  88.91 ms |
| 10 |  82.67 ms |
| 8 (number of my machine cores) | 88.49 ms |
| 6 |  87.23 ms |
| 5 | 86.54 ms |
| 2 |  112.17 ms |
| without | 128 ms |

I do not understand

btw, how is run `CherryPy` on **production** ? purpose `here` is to be close to **production**

Comment (webknjaz) [reply python/cherrypy/config.yaml:7]: CherryPy doesn't dictate best practices, it's a low-level minimalistic framework. It's very flexible and it's really up to users to decide how to deploy it.

Comment (murilobr) [reply python/cherrypy/config.yaml:7]: I think that in this case, this implementation need to extract the best from CherryPy.

Comment (waghanza) [reply python/cherrypy/config.yaml:7]: @murilobr @webknjaz I recommand to try with `10`, and we will see

**Comment by waghanza** [comment]
It's ok for me :heart:

File: python/cherrypy/config.yaml (hunk 1 above)
Comment (webknjaz) [reply python/cherrypy/config.yaml:7]: I think that this number is quite low but okay. Let's see how it goes. It depends on what load you're going to generate.

**Review by webknjaz — APPROVED** [review]

**Pull request merged**